source/ListingCommands.cpp

```diff
@@ -0,0 +1,293 @@
+  std::cout << "  • ContentProviderGenerator" << std::endl;
+  std::cout << "  • ServiceSourceGenerator" << std::endl;
+  std::cout << "  • TaintInTaintThisGenerator" << std::endl;
+  std::cout << "  • TaintInTaintOutGenerator" << std::endl;
+  std::cout << "  • BuilderPatternGenerator" << std::endl;
+  std::cout << "  • JoinOverrideGenerator" << std::endl;
+  std::cout << "  • ManifestSourceGenerator" << std::endl;
+  std::cout << "  • DFASourceGenerator" << std::endl;
+}
+
+void ListingCommands::list_all_kinds(Context& context) {
+  std::cout << "=== All Kinds ===" << std::endl;
+  
+  std::set<std::string> all_kinds;
+  
+  // Extract kinds from loaded rules
+  if (context.rules) {
+    for (const auto* rule : *context.rules) {
+      if (auto* source_sink_rule = rule->as<SourceSinkRule>()) {
+        for (const auto* source : source_sink_rule->source_kinds()) {
+          all_kinds.insert(source->to_trace_string());
+        }
+        for (const auto* sink : source_sink_rule->sink_kinds()) {
+          all_kinds.insert(sink->to_trace_string());
+        }
+      } else if (auto* exploitability_rule = rule->as<SourceSinkWithExploitabilityRule>()) {
+        for (const auto* effect_source : exploitability_rule->effect_source_kinds()) {
+          all_kinds.insert(effect_source->to_trace_string());
+        }
+        for (const auto* source : exploitability_rule->source_kinds()) {
+          all_kinds.insert(source->to_trace_string());
+        }
+        for (const auto* sink : exploitability_rule->sink_kinds()) {
+          all_kinds.insert(sink->to_trace_string());
+        }
+      } else if (auto* multi_rule = rule->as<MultiSourceMultiSinkRule>()) {
+        for (const auto& [label, kinds] : multi_rule->multi_source_kinds()) {
+          for (const auto* kind : kinds) {
+            all_kinds.insert(kind->to_trace_string());
+          }
+        }
+      }
+    }
+  }
+  
+  // Extract kinds from model generator files using JsonModelGenerator API
+  auto json_generator_paths = ModelGeneration::get_json_model_generator_paths(context);
+  for (const auto& [name, path] : json_generator_paths) {
+    try {
+      // Load model generator and extract kinds from generated models
+      auto generator = JsonModelGenerator::from_file(
+          name->identifier(), context, path);
+      
+      // Get models from the generator and extract kinds
+      auto models = generator.emit_method_models(*context.methods);
+      for (const auto& model : models) {
+        // Extract source kinds
+        for (const auto* kind : model.source_kinds()) {
+          all_kinds.insert(kind->to_trace_string());
+        }
+        // Extract sink kinds  
+        for (const auto* kind : model.sink_kinds()) {
+          all_kinds.insert(kind->to_trace_string());
+        }
+      }
+    } catch (const std::exception& e) {
+      std::cout << fmt::format("  Error loading generator {}: {}", path.string(), e.what()) << std::endl;
+    }
+  }
```

Comment:
I don't think listing sources and sinks from models is useful, since those should be the same as the ones from rules. I would think that listing kinds from rules is enough? Maybe we can only keep one of `--list-all-kinds` and `--list-all-kinds-from-rules`?

Also: please don't use `emit_method_models` as it's quite expensive. You should be able to get sources and sinks by iterating on `JsonModelGenerator::items_` and looking at `JsonModelGeneratorItem::model_template_`. But again, I would just recommend listing kinds from rules.